examples/solace/src/test/java/com/example/SolaceContainerTest.java

```diff
@@ -0,0 +1,443 @@
+                        LOGGER.error("Exception received: " + e.getMessage());
+                        latch.countDown();
+                    }
+                }
+            );
+            session.addSubscription(TOPIC);
+            cons.start();
+            publishMessageToSolace(session);
+            Assertions.assertThat(latch.await(10L, TimeUnit.SECONDS)).isTrue();
+            return result[0];
+        } catch (Exception e) {
+            throw new RuntimeException("Cannot receive message from solace", e);
+        }
+    }
+
+    static class SolaceContainer extends GenericContainer<SolaceContainer> {
+
+        private static final String NEW_LINE = System.getProperty("line.separator");
+
+        private static final String DEFAULT_VPN = "default";
+
+        private static final String DEFAULT_USERNAME = "default";
+
+        private static final String SOLACE_READY_MESSAGE = ".*Running pre-startup checks:.*";
+
+        private static final String SOLACE_ACTIVE_MESSAGE = "Primary Virtual Router is now active";
+
+        private static final String TMP_SCRIPT_LOCATION = "/tmp/script.cli";
+
+        private static final Long SHM_SIZE = (long) Math.pow(1024, 3);
+
+        private String username = "root";
+
+        private String password = "password";
+
+        private String vpn = DEFAULT_VPN;
+
+        private final List<Pair<String, Protocol>> topicsConfiguration = new ArrayList<>();
+
+        private boolean withClientCert;
+
+        public SolaceContainer(final DockerImageName dockerImageName) {
+            super(dockerImageName.toString());
+            dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+            this.waitStrategy = Wait.forLogMessage(SOLACE_READY_MESSAGE, 1).withStartupTimeout(Duration.ofSeconds(60));
+        }
+
+        @Override
+        protected void configure() {
+            withCreateContainerCmdModifier(cmd -> {
+                cmd
+                    .getHostConfig()
+                    .withShmSize(SHM_SIZE)
+                    .withUlimits(new Ulimit[] { new Ulimit("nofile", 2448L, 6592L) });
+            });
+            configureSolace();
+        }
+
+        private void configureSolace() {
+            withCopyToContainer(createConfigurationScript(), TMP_SCRIPT_LOCATION);
+        }
+
+        @Override
+        protected void containerIsStarted(InspectContainerResponse containerInfo) {
+            if (withClientCert) {
+                executeCommand("cp", "/tmp/solace.pem", "/usr/sw/jail/certs/solace.pem");
+                executeCommand("cp", "/tmp/rootCA.crt", "/usr/sw/jail/certs/rootCA.crt");
+            }
+            executeCommand("cp", TMP_SCRIPT_LOCATION, "/usr/sw/jail/cliscripts/script.cli");
+            waitOnCommandResult(
+                SOLACE_ACTIVE_MESSAGE,
+                "grep",
+                "-R",
+                SOLACE_ACTIVE_MESSAGE,
+                "/usr/sw/jail/logs/system.log"
```

Comment:
wonder if system.log is displayed as part of the container log. If so, checking against the container log would be easier.

Reply:
Sadly - it is not displayed. There is no information about that in the logs.